processing/src/main/java/org/apache/druid/segment/virtual/ExpressionSelectors.java

```diff
@@ -108,7 +108,11 @@ public Object getObject()
       {
         // No need for null check on getObject() since baseSelector impls will never return null.
         //noinspection ConstantConditions
-        return baseSelector.getObject().value();
+        ExprEval eval = baseSelector.getObject();
+        if (eval.isArray()) {
+          return Arrays.stream(eval.asArray()).map(String::valueOf).collect(Collectors.toList());
```

Comment:
Do we allow null numbers in lists? If so, this'll turn them into `"null"` rather than keeping them as `null`. What's the right behavior?

Reply:
Oops, probably not `"null"`. There are a few other places are doing this as well, will fix and add some tests around this behavior.